web_external/js/views/widgets/MeanContourWidget.js

```diff
@@ -64,18 +78,61 @@ minerva.views.MeanContourWidget = minerva.View.extend({
                     girder.events.trigger('m:job.created');
                 }, this));
             }, this));
+        },
+        'change #m-geojs-contour-input-source': function (evt) {
+            var el = $(evt.currentTarget);
+            var source = this.sourceCollection.get(el.val());
+            this._createSourceHierarchy(source);
         }
     },
 
     initialize: function (settings) {
-        this.datasetsCollection = settings.datasetsCollection;
+        this.datasetCollection = settings.datasetCollection;
+        this.sourceCollection = settings.sourceCollection;
         this.analysis = settings.analysis;
     },
 
+    _createSourceHierarchy: function (source) {
+        if (!source) {
+            this.$('.g-validation-failed-message').text('You must define at least one S3 source');
+            this.$('button.m-run-geojs-contour').addClass('disabled');
+            return;
+        }
+
+        this.folder = new girder.models.FolderModel();
+        var mm = source.mM();
+        this.folder.set({
+            _id: mm.folder_id,
+            minerva: mm
+        });
+
+        this.folder.on('g:fetched', function () {
+            this.$('.g-validation-failed-message').text('');
+            this.hierarchyWidget = new minerva.views.ReadOnlyHierarchyWidget({
+                el: $('.m-hierarchy-container'),
+                parentView: this,
+                parentModel: this.folder,
+                folderAccess: false,
+                upload: false,
+                folderCreate: false,
+                folderEdit: false,
+                itemCreate: false
+            }).render();
+        }, this).on('g:error', function () {
+            // TODO
+            console.error('fetch of folder failed');
+        }, this).fetch();
+    },
+
```

Comment:
I can see this function being useful in other widgets - maybe we should start thinking about creating a generic AnalysisWidget? #separateIssue